brownie/network/contract.py

```diff
@@ -127,6 +130,153 @@ def decode_input(self, calldata: Union[str, bytes]) -> Tuple[str, Any]:
+                                    symbol_alias["local"]
+                                )
+
+                offset = slice(*build_json["offset"])
+                used_offsets[build_json["source"]].append(tuple(build_json["offset"]))
+                source = self._slice_source(build_json["source"], offset)
+                code_fragments[name] = [source, name]
+                if not has_abiencoder:
+                    has_abiencoder = (
+                        len(re.findall(abiencoder_re, build_json["source"][: offset.start])) > 0
+                    )
+
+            for name in toposort_flatten(dependency_tree):
+                if name == "__root_node__":
+                    continue
+                source, part_name = code_fragments[name]
+                # Create a duplicate entry for each import alias
+                part_name_short = part_name
+                if "/" in part_name:
+                    part_name_short = part_name.split("/")[-1]
+                if part_name_short in import_aliases.keys():
+                    for a in import_aliases[part_name_short]:
+                        new_source = re.sub(
+                            rf"(library|contract|interface)\s+({part_name_short})",
+                            rf"\1 {a}",
+                            source,
+                        )
+                        flattened_source = f"{flattened_source}\n\n// Part: {a}\n\n{new_source}"
+                # Add the source without alias
+                flattened_source = f"{flattened_source}\n\n// Part: {part_name}\n\n{source}"
+
+            # Top level contract, defines compiler and license
+            build_json = self._build
+            version = build_json["compiler"]["version"]
+            version_short = re.findall(r"^[^+]+", version)[0]
+            offset = slice(*build_json["offset"])
+            used_offsets[build_json["source"]].append(tuple(build_json["offset"]))
+            source = self._slice_source(build_json["source"], offset)
+            file_name = Path(build_json["sourcePath"]).parts[-1]
+            licenses = re.findall(
+                r"SPDX-License-Identifier:(.*)\n", build_json["source"][: offset.start]
+            )
+            license_identifier = licenses[0].strip() if len(licenses) == 1 else "NONE"
+            if not has_abiencoder:
+                has_abiencoder = (
+                    len(re.findall(abiencoder_re, build_json["source"][: offset.start])) > 0
+                )
+
+            # find global structs and enums
+            global_structs = set()
+            global_enums = set()
+            for src, offsets in used_offsets.items():
+                offsets = sorted(offsets)
+                offsets.append((len(src), len(src)))
+                start = 0
+                for o in offsets:
+                    cleaned_section = remove_comments(src[start : o[0]])
+                    global_structs.update(re.findall(r"(\w*struct [^}]+})", cleaned_section))
```

Comment:
It would be better to find these using the AST.  Regex is too prone to false-positives...  someone might have a commented out struct or enum.